pages/dug_10/x-dasharo-dbx-uc-ci.md

```diff
@@ -0,0 +1,196 @@
+* **Firmware Chain of Trust:**
+  - Microcode -> coreboot -> UEFI -> Bootloader -> OS.
+  - Each stage verifies the next.
+
+<center><img src="/../../img/dug_10/dasharo_boot_diagram.png" width="900"></center>
+
+---
+
+## Our Goal: Consistent & Timely Security
+
+* Ensure critical components (Microcode, DBX) are **always** up-to-date.
+* Reduce risk of oversight.
+* Free up developer resources.
+* **Solution:** Automation!
+
+---
+
+## Component Deep Dive: Intel Microcode
+
+* **Source:** [Intel's GitHub](https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files)
+* **Purpose:** Fix security advisories, functional issues, and errata.
+  - Often essential for basic CPU operation.
+* **Loading:**
+  - By coreboot via Firmware Interface Table (FIT) – _before coreboot itself runs!_
+  - OS can load updates, but firmware-level is earlier & more comprehensive.
+* **Example Vulnerability Patched:**
+  - **INTEL-SA-01139:** Privilege escalation via UEFI module input validation.
+
+---
+
+## Component Deep Dive: UEFI DBX
+
+* **Source:** [UEFI Forum](https://uefi.org/revocationlistfile) / [Microsoft GitHub](https://github.com/microsoft/secureboot_objects)
+  - We use Microsoft's Git repo for easier automation & prompt updates.
+  - Discrepancy highlights need for reliable, scriptable sources.
+* **Purpose:** Revokes signatures of previously approved firmware/software.
+  - Critical for mitigating bootkits (e.g., BlackLotus) and compromised bootloaders.
+  - Revoking certificates is powerful: invalidates many binaries with one update.
+* **Example Vulnerability Mitigated:**
+  - **CVE-2022-21894 (BlackLotus):** Persistent UEFI bootkit bypassing Secure Boot.
+
+---
+
+## The Automation: GitHub Actions
+
+* **Why GitHub Actions?**
+  - Tight integration with our GitHub codebase.
+  - Declarative YAML syntax.
+  - Wide range of community-supported actions (e.g., for creating PRs).
+* Daily checks for updates to Microcode and DBX.
+
+<center><img src="/../../img/dug_10/gha_logo.png" width="150"></center>
+
+---
+
+## Automation: Microcode Update Workflow
+
+### 1. **Check Daily:**
+* Checkout Dasharo code & `intel-microcode` submodule.
+* Compare current submodule commit (`current`) with latest `main` branch of `intel-microcode` (`new`).
+* If `current != new`, submodule is outdated -> `exit 1`.
+
+```yaml
+# Microcode Check Snippet
+- name: Check if µcode submodule is up to date
+  run: |
+    git submodule update --init --checkout 3rdparty/intel-microcode
+    pushd 3rdparty/intel-microcode
+    current=<span class="math-inline">\(git log \-1 \-\-pretty\=format\:"%H"\)
+git checkout main \# Switch to main to get the latest
+new\=</span>(git log -1 --pretty=format:"%H")
+    if [[ $current == $new ]]; then
+      echo "Intel µcode submodule is up-to-date."
+    else
+      echo "Intel µcode submodule is out of date!"
+      exit 1
+    fi
+    popd
+```
+
+---
+
+## Automation: Microcode Update Workflow (Cont.)
```

Comment:
Draw a procedure diagram and on each slide or at each description step, show by dedicated graphics or mark with drawing tools where we are in your description

Reply:
Added, for ucode first: https://github.com/Dasharo/presentations/pull/94/commits/d804956b98392945b9910fb7dd95373f6820b81b

Reply:
dbx also done: https://github.com/Dasharo/presentations/pull/94/commits/0484b0d94eb699c9edc4b16bee41f1442ade71c2